src/charm.py

```diff
@@ -116,17 +123,16 @@ def _get_relation_data_or_config(self, field, fallback):
         return fallback
 
     @property
-    def _additional_hostnames(self) -> List[str]:
+    def _additional_hostnames(self) -> Generator[str, None, None]:
         """Return a list with additional hostnames.
 
         Returns:
             The additional hostnames set by configuration already split by comma.
         """
-        return [
-            x
-            for x in self._get_config_or_relation_data("additional-hostnames", "").split(",")
-            if x
-        ]
+        additional_hostnames = self._get_config_or_relation_data("additional-hostnames", "")
+        for hostname in additional_hostnames.split(","):
```

Comment:
You could do this by writing: `yield from filter(None, additional_hostnames.split(","))`

Reply:
Or yield from `(hostname for hostname in additional_hostnames if hostname)` depending on preference

Reply:
Changed, thanks